### ISSUE TITLE
Crunchyroll Beta - Currently Incompatible

### ISSUE THREAD

**Issue by noah978** [sttz/TraktRoller#21]
Since the new beta site of crunchyroll has been released this script no longer works for any US user that is logged into their crunchyroll account. You can always log out to get to the old website but that comes with its own set of downsides.

Ideally, this script should be updated to handle both the old site and the new one. (Since they are essentially two entirely different versions of the same website but both are still available to the public.)

From what I can tell, it should be a fairly simplistic fix, you just need to alter the loadPlayer() function to actually get the correct iframe since they changed the id tagging on the new website.

**Comment by sttz** [comment]
Yeah, I'm aware and have been working slowly on an update. Unfortunately, the new page is now generated by JavaScript. This makes injecting elements and scraping the episode information more complicated, since the HTML isn't there from the start and the script needs to wait for things to be generated/updated.

**Comment by pepijnmm** [comment]
most data get indeed later loaded but not the title/chapter. or which info do you need @sttz  ?
document.querySelector("[property='og:title']") for example gives the title and the episode when split in two
document.querySelector("[property='og:title']").content.split(" | ") you will get it nicely seperated.

what i did in an extention I ones made was just make a settimeout loop that starts the function it's in after 1000 seconds till it find the info it needs. not the nicest solution but users won't notice a document queryselect per second and then i stop doing the loop when i have all the data.
